From e9b73ac8c0e8550486e24f01c07268fd8be2f281 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Thu, 24 Jul 2025 10:34:10 +0100
Subject: [PATCH] quark: link against arch specific fmemopen symbol

Minimal, lowest, most portable fmemopen versioned symbols are
different on per arch basis.
---
 quark.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/quark.go b/quark.go
index 0dda63e..2fd75dd 100644
--- a/quark.go
+++ b/quark.go
@@ -13,7 +13,13 @@ package quark
    #include <stdlib.h>
    #include "quark.h"
 
+   #ifdef __x86_64__
    __asm__(".symver fmemopen, fmemopen@GLIBC_2.2.5");
+   #elif __aarch64__
+   __asm__(".symver fmemopen, fmemopen@GLIBC_2.17");
+   #else
+   #error Add correct desired symbol version for your arch
+   #endif
 
    FILE *
    __wrap_fmemopen(void *buf, size_t size, const char *mode)
-- 
2.48.1

